RealmTasks Apple/RealmTasks macOS/ListViewController.swift

```diff
@@ -349,26 +360,51 @@ final class ListViewController<ListType: ListPresentable where ListType: Object>
         view.window?.makeFirstResponder(cellView.textView)
 
         currentlyEditingCellView = cellView
+
+        beginUIWrite()
     }
 
     private func endEditingCells() {
-        guard editing else {
+        guard
+            let cellView = currentlyEditingCellView,
+            let (_, index) = findItemForCellView(cellView)
+        else {
             return
         }
 
+        var item = list.items[index]
+
+        if cellView.text != item.text || cellView.text.isEmpty {
+            if !cellView.text.isEmpty {
+                item.text = cellView.text
+            } else {
+                item.realm!.delete(item)
+
+                dispatch_async(dispatch_get_main_queue()) {
```

Comment:
why does this dispatch async to the main queue? AFAICT `endEditingCells()` should always be called from the main queue.

Is the goal here to perform the operation on the next iteration of the runloop? If so, a comment explaining that would be nice!